lib/jxl/render_pipeline/write_to_ib_stage.cc

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+#include "lib/jxl/render_pipeline/write_to_ib_stage.h"
+
+#include "lib/jxl/common.h"
+#include "lib/jxl/image_bundle.h"
+
+namespace jxl {
+
+namespace {
+class WriteToImageBundleStage : public RenderPipelineStage {
+ public:
+  explicit WriteToImageBundleStage(ImageBundle* image_bundle)
+      : RenderPipelineStage(RenderPipelineStage::Settings()),
+        image_bundle_(image_bundle) {}
```

Comment:
Might be useful to have some asserts/checks here to ensure some things:

- image bundle dimensions are OK (might also be useful to clarify what is expected in terms of orientation and padding to multiples of 8)
- image bundle doesn't have _more_ extra channels than what is available
- image bundle buffers are already allocated

(or possibly it might make sense / be more convenient to not just check but also do the correct initialization of the image_bundle here)

Reply:
I added some asserts.

FWIW orientation is not yet handled by any pipeline stage, so that's a non-concern for now.